packages/@react-types/shared/src/dnd.d.ts

```diff
@@ -142,7 +175,31 @@ export interface DroppableCollectionProps {
   /** Handler that is called when a valid drag exits the drop target. */
   onDropExit?: (e: DroppableCollectionExitEvent) => void,
   /** Handler that is called when a valid drag is dropped on the drop target. */
-  onDrop?: (e: DroppableCollectionDropEvent) => void
+  onDrop?: (e: DroppableCollectionDropEvent) => void,
+  /**
+   * Handler called when external items are dropped "between" the droppable collection's items.
+   */
+  onInsert?: (e: DroppableCollectionInsertDropEvent) => void,
+  /**
+   * Handler called when external items are dropped on the droppable collection's root.
+   */
+  onRootDrop?: (e: DroppableCollectionRootDropEvent) => void,
+  /**
+   * Handler called when items are dropped "on" a droppable collection's item.
+   */
+  onItemDrop?: (e: DroppableCollectionOnItemDropEvent) => void,
+  /**
+   * Handler called when items are reordered via drag in the source collection.
+   */
+  onReorder?: (e: DroppableCollectionReorderEvent) => void,
+  /**
+   * The drag types that the droppable collection accepts. If your collection accepts directories, include 'directory' in your array of allowed types.
+   */
+  acceptedDragTypes?: 'all' | Array<string>,
```

Comment:
I saw `'file'` in some of the examples too. Is that also a special type?

One thing I was thinking also was that this means you cannot have a custom type named `'directory'`. For example, we used `'folder'` in some of the examples for that. Is that confusing? Do we need another way to denote these special types?

Reply:
I haven't encountered a mime type of `'file'` out in the wild so I wouldn't consider a special type, but my testing isn't exhaustive by any means. Are you suggesting that we add a catch all option for files sourced from the finder/explorer similar to what we are doing with directory right now?

As for the `'directory'` type, I can definitely see that being a custom type used by external users so perhaps we should rename it to something more specific. Maybe `nativeDirectory`/`applicationDirectory`/`OSDirectory` or something along the lines of that?

Reply:
we could potentially use a [Symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) that we export to ensure there is no ambiguity...

Reply:
So something like `export const directory = Symbol('directory');` which users would import and then provide as part of the `acceptedDragTypes` array (e.g. `acceptedDragTypes={[directory, 'dragType1', 'text/plain']}`)? I can give that shot